erpnext/hr/doctype/employee_advance/employee_advance.js

```diff
@@ -138,16 +143,69 @@ frappe.ui.form.on('Employee Advance', {
 
 	employee: function (frm) {
 		if (frm.doc.employee) {
-			return frappe.call({
-				method: "erpnext.hr.doctype.employee_advance.employee_advance.get_pending_amount",
+			frappe.run_serially([
+				() => 	frm.trigger('get_employee_currency'),
+				() => 	frm.trigger('get_pending_amount')
+			]);
+		}
+	},
+
+	get_pending_amount: function(frm) {
+		frappe.call({
+			method: "erpnext.hr.doctype.employee_advance.employee_advance.get_pending_amount",
+			args: {
+				"employee": frm.doc.employee,
+				"posting_date": frm.doc.posting_date
+			},
+			callback: function(r) {
+				frm.set_value("pending_amount",r.message);
+			}
+		});
+	},
+
+	get_employee_currency: function(frm) {
+		frappe.call({
+			method: "erpnext.payroll.doctype.salary_structure_assignment.salary_structure_assignment.get_employee_currency",
+			args: {
+				employee: frm.doc.employee,
+			},
+			callback: function(r) {
+				if(r.message) {
+					frm.set_value('currency', r.message);
+					frm.refresh_fields();
+				}
+			}
+		});
+	},
+
+	currency: function(frm) {
+		var from_currency = frm.doc.currency;
+		var company_currency;
+		if (!frm.doc.company) {
+			company_currency = erpnext.get_currency(frappe.defaults.get_default("Company"));
+		}
+		else {
+			company_currency = erpnext.get_currency(frm.doc.company);
+		}
+		if(from_currency != company_currency) {
+			frappe.call({
+				method: "erpnext.setup.utils.get_exchange_rate",
 				args: {
-					"employee": frm.doc.employee,
-					"posting_date": frm.doc.posting_date
+					from_currency: from_currency,
+					to_currency: company_currency,
 				},
 				callback: function(r) {
-					frm.set_value("pending_amount",r.message);
+					frm.set_value("exchange_rate", flt(r.message));
+					frm.set_df_property('exchange_rate', 'hidden', 0);
+					frm.set_df_property("exchange_rate", "description", "1 " + frm.doc.currency
+						+ " = [?] " + company_currency);
 				}
 			});
+		} else {
+			frm.set_value("exchange_rate", 1.0);
+			frm.set_df_property('exchange_rate', 'hidden', 1);
+			frm.set_df_property("exchange_rate", "description", "" );
 		}
```

Comment:
split into a smaller function say set exchange rate. will be easy for debugging.